app/controllers/EventReportController.scala

```diff
@@ -64,24 +64,27 @@ class EventReportController @Inject()(
 
       withAuth.flatMap { case Credentials(externalId, psaOrPspId, _) =>
         val userAnswersJson = requiredBody.validate[JsObject].getOrElse(throw new RuntimeException("Expected JsObject body"))
-        val pstr = requiredHeaders("pstr").head
-        val etVersionYear = (request.headers.get("eventType"), request.headers.get("version"), request.headers.get("year")) match {
-          case (Some(et), Some(version), Some(year)) => Some((et, version.toInt, year.toInt))
-          case _ => None
-        }
+        requiredHeader("pstr") match {
+          case Left(msg) => Future.successful(BadRequest(msg))
+          case Right(pstr) =>
+            val etVersionYear = (request.headers.get("eventType"), request.headers.get("version"), request.headers.get("year")) match {
+              case (Some(et), Some(version), Some(year)) => Some((et, version.toInt, year.toInt))
+              case _ => None
```

Comment:
could we log.warn that it failed to get some headers here

Reply:
I think this is intended behaviour for non event type user answers.